File: ArchitectureRefactoringHelper/Repository/Models/Recommendation/ApproachRecommendation.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;
using Repository.Models.Database;

namespace Repository.Models.Recommendation;

public class ApproachRecommendation
{
    [Required]
    [JsonPropertyName("refactoringApproachId")]
    public int RefactoringApproachId { get; set; }
    [Required]
    [JsonPropertyName("identifier")]
    public string Identifier { get; set; }

    [Required]
    [JsonPropertyName("approachSource")]
    public ApproachSource ApproachSource { get; set; }

    [JsonPropertyName("domainArtifactInputEvaluations")]
    public IList<ApproachAttributeEvaluation<DomainArtifactInput>> DomainArtifactInputEvaluations { get; set; }

    [JsonPropertyName("runtimeArtifactInputEvaluations")]
    public IList<ApproachAttributeEvaluation<RuntimeArtifactInput>> RuntimeArtifactInputEvaluations { get; set; }

    [JsonPropertyName("modelArtifactInputEvaluations")]
    public IList<ApproachAttributeEvaluation<ModelArtifactInput>> ModelArtifactInputEvaluations { get; set; }

    [JsonPropertyName("executableInputEvaluations")]
    public IList<ApproachAttributeEvaluation<ExecutableInput>> ExecutableInputEvaluations { get; set; }

    [JsonPropertyName("qualityEvaluations")]
    public IList<ApproachAttributeEvaluation<Quality>> QualityEvaluations { get; set; }
    [JsonPropertyName("qualitySublevelEvaluations")]
    public IList<ApproachAttributeEvaluation<QualitySublevel>> QualitySublevelEvaluations { get; set; }

    [JsonPropertyName("directionEvaluations")]
    public IList<ApproachAttributeEvaluation<Direction>> DirectionEvaluations { get; set; }

    [JsonPropertyName("automationLevelEvaluations")]
    public IList<ApproachAttributeEvaluation<AutomationLevel>> AutomationLevelEvaluations { get; set; }

    [JsonPropertyName("analysisTypeEvaluations")]
    public IList<ApproachAttributeEvaluation<AnalysisType>> AnalysisTypeEvaluations { get; set; }

    [JsonPropertyName("techniqueEvaluations")]
    public IList<ApproachAttributeEvaluation<Technique>> TechniqueEvaluations { get; set; }

    [JsonPropertyName("architectureEvaluations")]
    public IList<ApproachAttributeEvaluation<Architecture>> ArchitectureEvaluations { get; set; }

    [JsonPropertyName("serviceTypeEvaluations")]
    public IList<ApproachAttributeEvaluation<ServiceType>> ServiceTypeEvaluations { get; set; }

    [JsonPropertyName("validationMethodEvaluations")]
    public IList<ApproachAttributeEvaluation<ValidationMethod>> ValidationMethodEvaluations { get; set; }

    [JsonPropertyName("toolSupportEvaluations")]
    public IList<ApproachAttributeEvaluation<ToolSupport>> ToolSupportEvaluations { get; set; }

    [JsonPropertyName("resultsQualityEvaluations")]
    public IList<ApproachAttributeEvaluation<ResultsQuality>> ResultsQualityEvaluations { get; set; }

    [JsonPropertyName("accuracyPrecisionEvaluations")]
    public IList<ApproachAttributeEvaluation<AccuracyPrecision>> AccuracyPrecisionEvaluations { get; set; }

    [Required]
    [JsonPropertyName("suitabilityScore")]
    public int SuitabilityScore { get; set; }

    [Required]
    [JsonPropertyName("matchesCount")]
    public int MatchesCount { get; set; }

    [Required]
    [JsonPropertyName("totalIncludeCount")]
    public int TotalIncludeCount { get; set; }

    [Required]
    [JsonPropertyName("qualityScore")]
    public QualityScore? QualityScore { get; set; }

    [Required]
    [JsonPropertyName("systemPropertiesScore")]
    public SystemPropertiesScore? SystemPropertiesScore { get; set; }

    [Required]
    [JsonPropertyName("totalScore")]
    public int TotalScore { get; set; }

    [Required]
    [JsonPropertyName("weightedScore")]
    public int WeightedScore { get; set; }

}

File: UI/api/repository/models/approach-recommendation.ts
/* tslint:disable */
/* eslint-disable */
import { AccuracyPrecisionApproachAttributeEvaluation } from './accuracy-precision-approach-attribute-evaluation';
import { AnalysisTypeApproachAttributeEvaluation } from './analysis-type-approach-attribute-evaluation';
import { ApproachSource } from './approach-source';
import { ArchitectureApproachAttributeEvaluation } from './architecture-approach-attribute-evaluation';
import { AutomationLevelApproachAttributeEvaluation } from './automation-level-approach-attribute-evaluation';
import { DirectionApproachAttributeEvaluation } from './direction-approach-attribute-evaluation';
import { DomainArtifactInputApproachAttributeEvaluation } from './domain-artifact-input-approach-attribute-evaluation';
import { ExecutableInputApproachAttributeEvaluation } from './executable-input-approach-attribute-evaluation';
import { ModelArtifactInputApproachAttributeEvaluation } from './model-artifact-input-approach-attribute-evaluation';
import { QualityApproachAttributeEvaluation } from './quality-approach-attribute-evaluation';
import { QualityScore } from './quality-score';
import { QualitySublevelApproachAttributeEvaluation } from './quality-sublevel-approach-attribute-evaluation';
import { ResultsQualityApproachAttributeEvaluation } from './results-quality-approach-attribute-evaluation';
import { RuntimeArtifactInputApproachAttributeEvaluation } from './runtime-artifact-input-approach-attribute-evaluation';
import { ServiceTypeApproachAttributeEvaluation } from './service-type-approach-attribute-evaluation';
import { SystemPropertiesScore } from './system-properties-score';
import { TechniqueApproachAttributeEvaluation } from './technique-approach-attribute-evaluation';
import { ToolSupportApproachAttributeEvaluation } from './tool-support-approach-attribute-evaluation';
import { ValidationMethodApproachAttributeEvaluation } from './validation-method-approach-attribute-evaluation';
export interface ApproachRecommendation {
  accuracyPrecisionEvaluations?: null | Array<AccuracyPrecisionApproachAttributeEvaluation>;
  analysisTypeEvaluations?: null | Array<AnalysisTypeApproachAttributeEvaluation>;
  approachSource: ApproachSource;
  architectureEvaluations?: null | Array<ArchitectureApproachAttributeEvaluation>;
  automationLevelEvaluations?: null | Array<AutomationLevelApproachAttributeEvaluation>;
  directionEvaluations?: null | Array<DirectionApproachAttributeEvaluation>;
  domainArtifactInputEvaluations?: null | Array<DomainArtifactInputApproachAttributeEvaluation>;
  executableInputEvaluations?: null | Array<ExecutableInputApproachAttributeEvaluation>;
  identifier: string;
  modelArtifactInputEvaluations?: null | Array<ModelArtifactInputApproachAttributeEvaluation>;
  qualityEvaluations?: null | Array<QualityApproachAttributeEvaluation>;
  qualityScore: QualityScore;
  qualitySublevelEvaluations?: null | Array<QualitySublevelApproachAttributeEvaluation>;
  refactoringApproachId: number;
  resultsQualityEvaluations?: null | Array<ResultsQualityApproachAttributeEvaluation>;
  runtimeArtifactInputEvaluations?: null | Array<RuntimeArtifactInputApproachAttributeEvaluation>;
  serviceTypeEvaluations?: null | Array<ServiceTypeApproachAttributeEvaluation>;
  suitabilityScore: number;
  systemPropertiesScore: SystemPropertiesScore;
  techniqueEvaluations?: null | Array<TechniqueApproachAttributeEvaluation>;
  toolSupportEvaluations?: null | Array<ToolSupportApproachAttributeEvaluation>;
  totalScore: number;
  validationMethodEvaluations?: null | Array<ValidationMethodApproachAttributeEvaluation>;
  weightedScore: number;
}
